src/Test/L0/Worker/TaskManagerL0.cs

```diff
@@ -843,6 +756,138 @@ private void Teardown()
+                {
+                    var bingGuid = Guid.NewGuid();
+                    string bingTaskName = "Bing";
+                    string bingVersion = "1.21.2";
+                    var tasks = new List<Pipelines.TaskStep>
+                    {
+                        new Pipelines.TaskStep()
+                        {
+                            Enabled = true,
+                            Reference = new Pipelines.TaskStepDefinitionReference()
+                            {
+                                Name = bingTaskName,
+                                Version = bingVersion,
+                                Id = bingGuid
+                            }
+                        },
+                        new Pipelines.TaskStep()
+                        {
+                            Enabled = true,
+                            Reference = new Pipelines.TaskStepDefinitionReference()
+                            {
+                                Name = bingTaskName,
+                                Version = bingVersion,
+                                Id = bingGuid
+                            }
+                        }
+                    };
+                    using (var stream = GetZipStream())
+                    {
+                        _taskServer
+                            .Setup(x => x.GetTaskContentZipAsync(
+                                bingGuid,
+                                It.Is<TaskVersion>(y => string.Equals(y.ToString(), bingVersion, StringComparison.Ordinal)),
+                                It.IsAny<CancellationToken>()))
+                            .Returns(Task.FromResult<Stream>(stream));
+
+                        //Act
+                        //first invocation will download and unzip the task from mocked IJobServer
+                        await _taskManager.DownloadAsync(_ec.Object, tasks);
+
+                        string destDirectory = Path.Combine(
+                            _hc.GetDirectory(WellKnownDirectory.Tasks),
+                            $"{bingTaskName}_{bingGuid}",
+                            bingVersion);
+ 
+                        //see if the task.json was downloaded
+                        string zipDestDirectory = Path.Combine(_hc.GetDirectory(WellKnownDirectory.TaskZips), $"{bingTaskName}_{bingGuid}_{bingVersion}.zip");
+
+                        // task.json should exist since we need it for JobExtension.InitializeJob
+                        Assert.True(File.Exists(Path.Combine(destDirectory, Constants.Path.TaskJsonFile)));
+
+                        // Write a test file
+                        string testFile = Path.Combine(destDirectory, "test.txt");
+                        using (File.Create(testFile)) { } 
+                        Assert.True(File.Exists(testFile));
+
+                        //second and third invocations should find the task in the cache and do nothing
```

Comment:
This comment is no longer accurate since we may download the task again?

Reply:
Good catch!